skbio/tree/_tree.py

```diff
@@ -3415,34 +3864,60 @@ def descending_branch_length(self, tip_subset=None):
                 if n.length is not None
             )
 
-    def distance(self, other):
+    def distance(self, other, use_length=True, missing_as_zero=False):
         """Calculate the distance between self and another node.
 
-        This method can be used to compute the distances between two tips,
-        however, it is not optimized for computing pairwise tip distances.
-
         Parameters
         ----------
         other : TreeNode
             The node to compute a distance to.
+        use_length : bool, optional
+            Whether to return the sum of branch lengths (True, default) or the number
+            of branches (False) connecting self and other.
+
+            .. versionadded:: 0.6.3
+
+        missing_as_zero : bool, optional
+            When a node without an associated branch length is encountered, raise an
+            error (False, default) or use 0 (True). Applicable when `length` is True.
+
+            .. versionadded:: 0.6.3
 
         Returns
         -------
-        float
+        float or int
```

Comment:
Same comment w/ return type. It is not incorrect to force this value to be `float`, but if `int` is important then I recommend considering expressing it as a distinct method so the function signature is fixed